src/cascadia/TerminalSettingsEditor/MainPage.cpp

```diff
@@ -122,8 +123,6 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         // refresh the current page using the SelectedItem data we collected before the refresh
         if (selectedItemTag)
         {
-            const auto& selectedItemStringTag{ selectedItemTag.try_as<hstring>() };
-            const auto& selectedItemProfileTag{ selectedItemTag.try_as<ProfileViewModel>() };

```

Comment:
Curious: were these changes necessary? I figured defining them outside of the for-loop would save a bit of time, but I did that because I honestly don't know how expensive doing `try_as` twice for each `item` was.

Reply:
If I recall (Pankaj can clear this up), we were treating the selected item _DIRECTLY_ as a string, or as a PVM, every time we hit a string or PVM in the Items list. This meant that when we were iterating over a *profile*, we would attempt to access `->Guid()` on the selected item (oops).

Reply:
That is: we automatically treated the selected item _as a profile_ if we happened to be iterating over a profile. That would _of course_ explode when the selected item was not a profile.

Reply:
Yep that's what was happening!